### PR TITLE
split: functionality of validators

### PR DESCRIPTION
**Pull request by joren-dev**
# Overview

## Changes:
Be precise and detailed.
- *Splits the functionality of SignUp and SignIn functionality into multiple parts*

## Any changes on files besides the ones directly connected to your feature/change (emtpy if none):
E.g. if you integrated your feature into another interface, said change should be listed here.
- *insert here*

## Additional Notes:
Anything else one should know about this pull request.
- *insert here*

</br>

# Extra
- [ ] Add explanation/information on wiki page

### CODE REVIEW

File: docs/nl/ai42/utils/validation/LoginValidator.html
Comment (deveolper) [inline docs/nl/ai42/utils/validation/LoginValidator.html:1]: SignInValidator.html voor consistentcy?

File: src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                                                 DatePicker datePicker) {
+        if (isEmptyField(usernameTextField) || isEmptyField(emailTextField) ||
+                isEmptyField(passwordField) || isEmptyField(repeatPasswordField) ||
+                datePicker.getValue() == null) {
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validateUsername(TextField usernameTextField) {
```
Comment (deveolper) [inline src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java:26]: Deze methode kan denk ik prima een one-liner zijn.

File: src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (!PatternMatchUtility.isValidName(usernameTextField.getText())) {
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validateEmail(TextField emailTextField) {
```
Comment (deveolper) [inline src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java:30]: Deze methode kan denk ik ook een one-liner zijn.

File: src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        if (!PatternMatchUtility.isValidEmail(emailTextField.getText())) {
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validatePassword(TextField passwordField) {
```
Comment (deveolper) [inline src/main/java/nl/ai42/utils/validation/utility/ValidationUtility.java:34]: Same for this one.

**Review by deveolper — CHANGES_REQUESTED** [review]
Zie comments

File: docs/nl/ai42/utils/validation/LoginValidator.html
Comment (joren-dev) [reply docs/nl/ai42/utils/validation/LoginValidator.html:1]: Updated website fully

**Pull request merged**